qiita_pet/util.py

```diff
@@ -116,3 +116,45 @@ def is_localhost(host):
     """
     localhost = ('localhost', '127.0.0.1')
     return host.startswith(localhost)
+
+
+def get_artifact_processing_status(artifact):
+    """Gets the processing status of the artifact
+
+    Parameters
+    ----------
+    artifact : qiita_db.artifact.Artifact
+        The artifact to get the processing status
+
+    Returns
+    -------
+    str, str
+        The processing status {'processing', 'failed', 'success',
+            'Not processed'}
+        A summary of the jobs attached to the artifact
+    """
+    preprocessing_status = 'Not processed'
+    preprocessing_status_msg = []
+    for job in artifact.jobs():
+        job_status = job.status
+        if job_status == 'error':
+            if preprocessing_status != 'success':
+                preprocessing_status = 'failed'
+            preprocessing_status_msg.append(
+                "<b>Job %s</b>: failed - %s"
+                % (job.id, job.log.msg))
+        elif job_status == 'success':
+            preprocessing_status = 'success'
```

Comment:
Just to sanity check

Wouldn't it be possible to have an artifact that has multiple jobs, the first job fails, but the last job succeeds, resulting in `get_artifact_status` to return `'sucess'`.

Is this behavior wanted?